libs/hwManager/manufacturers/trezor.js

```diff
@@ -46,24 +80,44 @@ const executeCommand = (transporter, {
   }
 
   return new Promise((resolve, reject) => {
+    // eslint-disable-next-line max-statements
     trezorDevice.waitForSessionAndRun(async (session) => {
-      switch (action) {
-        case 'GET_PUBLICKEY': {
-          const { message } = await session.typedCall(
-            'LiskGetPublicKey',
-            'LiskPublicKey',
-            {
-              address_n: getHardenedPath(data.index),
-              show_display: data.showOnDevice,
-            },
-          );
-          return resolve(message.public_key);
-        }
-        default: {
-          // eslint-disable-next-line no-console
-          console.log(`No action created for: ${device.manufactor}.${action}`);
-          return reject(new Error(`No action created for: ${device.manufactor}.${action}`));
+      try {
+        switch (action) {
```

Comment:
Why do we need `HwManager.executeCommand` and `switch (action)`? IMO a cleaner API would be to have a method for each command:`HwManager.getPublicKey`, `HwManager.signTx`... And it's a matter of how to implement these methods in a way that they are DRY and don't repeat the same logic. Otherwise, this `switch` will be overgrown.

Reply:
That was intended to keep backward compatibility due to the currents PRs only changing the `"provider side"` of the hwManager, but I agree that it could be done in a way that for now it would just forward the call to the correct hw manager function, and when the `"consumer side"` is updated it can communicate directly to the correct function.

Reply:
Ah, ok. Got it. 
Good to hear from you @massao, BTW 🙂